### PR TITLE
Kelsha Darby - User Story 4

### LINKED ISSUE #7: User Story 4, Merchant Navigation
As a merchant employee
I see the same links as a regular user
Plus the following links:
- a link to my merchant dashboard ("/merchant")

### PR DESCRIPTION
**Pull request by kelshadarby**
### Merchant Navigation Bar

_Test_
- Added to `spec/features/users/nav_bar_spec.rb` to test the merchant nav bar
     - I think these tests can be moved to the navigation bar test or maybe even just renamed but wanted to check with you all first.

_View_
- Added routes to application view that show depending on users role or lack thereof

_Controller_
- Added `current_user` to the `application_controller` helper methods so it can be accessed in the view

Closes: #7

### CODE REVIEW

File: spec/features/users/nav_bar_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+      visit "/default/profile"
+
+      expect(page).to have_link("Home")
+      expect(page).to have_link("All Merchants")
+      expect(page).to have_link("All Items")
+      expect(page).to have_link("Cart: 0")
+      expect(page).to have_link("View My Profile")
+      expect(page).to have_link("Logout")
+
```
Comment (brian-greeson) [inline spec/features/users/nav_bar_spec.rb]: We should refactor this to actually click on the links

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -10,14 +10,19 @@ def create
...
+      flash[:success] = "Logged in as #{user.user_detail.name}"
       dynamic_redirect
     else
       flash[:error] = "Invalid Credentials"
       render :new
     end
   end
 
+  def destroy
+    session[:user_id] = nil
```
Comment (fredrondina) [inline app/controllers/sessions_controller.rb:22]: nice job proactively adding this, I will add the test for US 16 after the git fire is extinguished

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -8,23 +8,25 @@ def create
...
       redirect_to profile_path
     else
       flash[:error] = @user.errors.full_messages.to_sentence
       render :new
     end
   end
-  
+
   def show
+    user = User.find(current_user.id)
```
Comment (fredrondina) [inline app/controllers/users_controller.rb:20]: nice job using this for the profile view

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -16,6 +16,9 @@
       <%= link_to "Cart: #{cart.total_items}", "/cart" %>
       <%= link_to "Register", register_path if !current_user %>
       <%= link_to "Login", login_path if !current_user %>
+      <%= link_to "View My Profile", "/profile" if current_user %>
+      <%= link_to "Dashboard", merchant_dashboard_path if current_user && current_user.role == "merchant" %>
```
Comment (fredrondina) [inline app/views/layouts/application.html.erb:20]: nice job using the name-spaced route

File: config/routes.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@
 
   get "/login", to: "sessions#new"
   post "/login", to: "sessions#create"
+  delete "/logout", to: "sessions#destroy"
```
Comment (fredrondina) [inline config/routes.rb:6]: I will add the feature test for this tonight

**Review by fredrondina — APPROVED** [review]
Awesome work all around, I like preemptively adding some functionality from US 16, its now complete besides the test, which I will add tonight.

**Pull request merged**